common/FileUtil.cpp

```diff
@@ -234,11 +234,24 @@ namespace FileUtil
     std::string getTempFileCopyPath(const std::string& srcDir, const std::string& srcFilename, const std::string& dstFilenamePrefix)
     {
         const std::string srcPath = srcDir + '/' + srcFilename;
-        const std::string dstFilename = dstFilenamePrefix + Util::encodeId(Util::rng::getNext()) + '_' + srcFilename;
 #if HAVE_STD_FILESYSTEM
-        // Don't const to allow for automatic move on return.
-        std::string dstPath = filesystem::temp_directory_path() / dstFilename;
-        filesystem::copy(srcPath, dstPath);
+        std::string dstPath;
+
+        bool retry;
+        do {
+            std::string dstFilename = dstFilenamePrefix + Util::encodeId(Util::rng::getNext()) + '_' + srcFilename;
+
+            retry = false;
+            dstPath = filesystem::temp_directory_path() / dstFilename;
+            try {
+                filesystem::copy(srcPath, dstPath);
+            }
+            catch (const std::exception& ex)
+            {
+                LOG_SYS("ERROR: unexpected conflict creating file: " << ex.what());
+                retry = true;;
+            }
+        } while (retry);
```

Comment:
are we asking for some sort of infinite loop if we run out of disk space or some other arbitrary error?

Reply:
Well - currently we throw an exception and crash; also that exception is a horror:

terminate called after throwing an instance of 'std::experimental::filesystem::v1::__cxx11::filesystem_error'

And not worth catching IMHO ;-) and TBH - I have really no idea why we use this trendy C++ thing instead of the other code-path that (IIRC) was more reliable anyway for random reasons in the past :-)

Anyhow - if we get the random piece in prolly we don't need this other bit nearly as much.

Reply:
I see getTempFileCopyPath is only used in tests so doesn't get used outside of there, so that concern doesn't matter a whole pile from its usage. Can move it into tests later maybe.